pkg/storage/chunk/client/object_client.go

```diff
@@ -183,6 +183,11 @@ func (o *client) getChunk(ctx context.Context, decodeContext *chunk.DecodeContex
 	}
 	defer readCloser.Close()
 
+	// reset if GetObject fails to return a valid size
+	if size < 0 {
+		size = 0
+	}
```

Comment:
hey would that be possible a bug on our side? If so, i think we should at least log something. If that's a known problem with the GCP API please ignore this comment.

Reply:
we log it in the called method [here](https://github.com/grafana/loki/pull/11132/files#diff-4ccb492979e70aa118915885936aa252cc68d2e3164a0dc681e511e8192b1432R102)

I added some context [here](https://github.com/grafana/loki/pull/11132#discussion_r1800962297) about the check, but this is unlikely to fail. I'll monitor for errors as i test this out